### PR TITLE
Code review

### PR DESCRIPTION
**Pull request by rodfersou**
- Don't automatically process old pages.
- Removed broken cache
- Provide an alternative to extract the page content for Archetypes

### CODE REVIEW

File: src/brasil/gov/vlibrasvideo/browser/vlibrasvideo.py [hunk 1]
```diff
@@ -25,20 +20,8 @@ def _is_ready(self):
         self.status, self.video_url = get_video_url(self.context)
         if self.status == 200:
             return True
-        elif self.status != 102 and _get_state(self.context) == 'published':
-            post_news(self.context)
         return False
 
-    @ram.cache(lambda *args: (time() // TTL))
```
Comment (hvelarde) [inline src/brasil/gov/vlibrasvideo/browser/vlibrasvideo.py]: @rodfersou just comment this line and leave the rest; we'll use this later

File: src/brasil/gov/vlibrasvideo/utils.py [hunk 2]
```diff
@@ -61,31 +58,47 @@ def _get_registry(name, default=''):
...
+    :param context: [required] Content object
+    :type context: content object
+    :param token: [required] Token used to access API
+    :type token: string
+    :returns: true if object is ready
+    :rtype: bool
+    """
+    # check if content type is enabled
+    enabled_content_types = _get_registry('enabled_content_types', [])
+    type_is_enabled = getattr(context, 'portal_type', '') in enabled_content_types
```
Comment (hvelarde) [inline src/brasil/gov/vlibrasvideo/utils.py]: check the condition here to abort early

File: src/brasil/gov/vlibrasvideo/utils.py [hunk 3]
```diff
@@ -61,31 +58,47 @@ def _get_registry(name, default=''):
...
+
+    # check if object is published
+    state = 'private'
+    try:
+        state = api.content.get_state(obj=context)
+    except WorkflowException, api.exc.CannotGetPortalError:
+        pass
+    is_published = state == 'published'
```
Comment (hvelarde) [inline src/brasil/gov/vlibrasvideo/utils.py]: check the condition here to abort early

File: src/brasil/gov/vlibrasvideo/utils.py [hunk 4]
```diff
@@ -21,26 +25,19 @@ def _get_content(obj):
...
     transformer = ITransformer(obj)
     text = getattr(obj, 'text', '')
-    if text:
+    if IRichTextValue.providedBy(text):
+        # Dexterity
         text = transformer(obj.text, 'text/plain')
+    elif getattr(obj, 'getText', False):
+        # Archetypes
+        data = obj.getText()
+        if data == '':
```
Comment (hvelarde) [inline src/brasil/gov/vlibrasvideo/utils.py]: I don't understand this

File: src/brasil/gov/vlibrasvideo/utils.py [hunk 5]
```diff
@@ -149,58 +153,52 @@ def repost_news(context, event=None):
...
-    response = requests.put(**params)
-    logger.info(u'{0}: {1} - {2}'.format(
-        context.absolute_url(), response.status_code, response.reason
-    ))
-    return (response.status_code, response.reason)
+    try:
+        response = requests.put(**params)
+    except RequestException as e:  # skip on timeouts and other errors
+        logger.error(u'{0}: {1}'.format(
+            context.absolute_url(), e.message
```
Comment (hvelarde) [inline src/brasil/gov/vlibrasvideo/utils.py]: I think you should log the URL of the remote server

File: src/brasil/gov/vlibrasvideo/utils.py [hunk 6]
```diff
@@ -149,58 +153,52 @@ def repost_news(context, event=None):
...
         headers=headers,
         timeout=REQUEST_TIMEOUT
     )
     logger.info('GET - {0}'.format(params))
-    response = requests.get(**params)
+    try:
+        response = requests.get(**params)
+    except RequestException as e:  # skip on timeouts and other errors
+        logger.error(u'{0}: {1}'.format(
+            context.absolute_url(), e.message
```
Comment (hvelarde) [inline src/brasil/gov/vlibrasvideo/utils.py]: I think you should log the URL of the remote server

File: src/brasil/gov/vlibrasvideo/utils.py [hunk 7]
```diff
@@ -149,58 +153,52 @@ def repost_news(context, event=None):
...
+        ))
+        return
     if response.status_code != 200:
-        logger.info(u'{0}: {1} - {2}'.format(
+        logger.error(u'{0}: {1} - {2}'.format(
             context.absolute_url(), response.status_code, response.reason
```
Comment (hvelarde) [inline src/brasil/gov/vlibrasvideo/utils.py]: I think you should log the URL of the remote server

File: src/brasil/gov/vlibrasvideo/utils.py [hunk 8]
```diff
@@ -209,38 +207,30 @@ def delete_video(context, event=None):
...
+    try:
+        response = requests.delete(**params)
+    except RequestException as e:  # skip on timeouts and other errors
+        logger.error(u'{0}: {1}'.format(
+            context.absolute_url(), e.message
+        ))
+        return False
+    if response.status_code != 200:
+        logger.error(u'{0}: {1} - {2}'.format(
+            context.absolute_url(), response.status_code, response.reason
```
Comment (hvelarde) [inline src/brasil/gov/vlibrasvideo/utils.py]: I think you should log the URL of the remote server

File: src/brasil/gov/vlibrasvideo/utils.py (hunk 4 above)
Comment (rodfersou) [inline src/brasil/gov/vlibrasvideo/utils.py]: this is to avoid an exception when run `html.fromstring('')`

Comment (hvelarde) [inline src/brasil/gov/vlibrasvideo/utils.py]: and what's the value of `text`?

Comment (rodfersou) [inline src/brasil/gov/vlibrasvideo/utils.py]: `''` I'll review the code for better reading.

File: src/brasil/gov/vlibrasvideo/utils.py (hunk 8 above)
Comment (rodfersou) [inline src/brasil/gov/vlibrasvideo/utils.py]: the URL of the remote server are always fixed, but the context where we got the exception is not.  I think just add a text saying that it is a POST is sufficient to know what is going on here.

File: src/brasil/gov/vlibrasvideo/utils.py (hunk 5 above)
Comment (rodfersou) [inline src/brasil/gov/vlibrasvideo/utils.py]: the URL of the remote server are always fixed, but the context where we got the exception is not. I think just add a text saying that it is a PUT is sufficient to know what is going on here.

File: src/brasil/gov/vlibrasvideo/utils.py (hunk 6 above)
Comment (rodfersou) [inline src/brasil/gov/vlibrasvideo/utils.py]: the URL of the remote server are always fixed, but the context where we got the exception is not. I think just add a text saying that it is a GET is sufficient to know what is going on here.

File: src/brasil/gov/vlibrasvideo/utils.py (hunk 7 above)
Comment (rodfersou) [inline src/brasil/gov/vlibrasvideo/utils.py]: the URL of the remote server are always fixed, but the context where we got the exception is not. I think just add a text saying that it is a GET is sufficient to know what is going on here.

File: src/brasil/gov/vlibrasvideo/utils.py (hunk 2 above)
Comment (rodfersou) [inline src/brasil/gov/vlibrasvideo/utils.py]: OK

File: src/brasil/gov/vlibrasvideo/utils.py (hunk 3 above)
Comment (rodfersou) [inline src/brasil/gov/vlibrasvideo/utils.py]: OK

**Comment by rodfersou** [comment]
@hvelarde I think it is ready now

**Comment by rodfersou** [comment]
Depends on: http://stackoverflow.com/questions/37883807/what-is-the-canonical-way-to-get-text-from-richtext-field-with-archetypes

**Pull request merged**